relay-profiling/src/cocoa.rs

```diff
@@ -121,11 +121,33 @@ struct CocoaProfile {
     version_name: String,
 }
 
+impl CocoaProfile {
+    fn filter_samples(&mut self) {
+        let mut sample_count_by_thread_id: HashMap<u64, u32> = HashMap::new();
+
+        for sample in self.sampled_profile.samples.iter() {
+            *sample_count_by_thread_id
+                .entry(sample.thread_id)
+                .or_default() += 1;
+        }
+
+        sample_count_by_thread_id.retain(|_, count| *count > 1);
+
+        let thread_ids: Vec<u64> = sample_count_by_thread_id.keys().cloned().collect();
+
+        self.sampled_profile
+            .samples
+            .retain(|sample| thread_ids.contains(&sample.thread_id));
```

Comment:
What do you think if instead of creating a vector with thread IDs we could just use something like [`contains_key`](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.contains_key) on HashMap? 

```suggestion
        self.sampled_profile
            .samples
            .retain(|sample| sample_count_by_thread_id.contains_key(&sample.thread_id));
```

Reply:
Oh, of course 🤦 .